src/zenml/materializers/base_materializer.py

```diff
@@ -51,61 +52,116 @@ def __new__(
         cls = cast(
             Type["BaseMaterializer"], super().__new__(mcs, name, bases, dct)
         )
-        if name != "BaseMaterializer":
-            from zenml.artifacts.base_artifact import BaseArtifact
 
-            if not cls.ASSOCIATED_TYPES:
+        # Skip the following validation and registration for the base class.
+        if name == "BaseMaterializer":
+            return cls
+
+        # Validate that the class is properly defined.
+        if not cls.ASSOCIATED_TYPES:
+            raise MaterializerInterfaceError(
+                f"Invalid materializer class '{name}'. When creating a "
+                f"custom materializer, make sure to specify at least one "
+                f"type in its ASSOCIATED_TYPES class variable.",
+                url="https://docs.zenml.io/advanced-guide/pipelines/materializers",
+            )
+
+        # Check if the deprecated ASSOCIATED_ARTIFACT_TYPES is used.
+        artifact_type: str = cls.ASSOCIATED_ARTIFACT_TYPE
+        if cls.ASSOCIATED_ARTIFACT_TYPES:
+            logger.warning(
+                "The ASSOCIATED_ARTIFACT_TYPES class variable is deprecated "
+                "and will be removed in a future release. Please use "
+                "ASSOCIATED_ARTIFACT_TYPE instead."
+            )
+            artifact_class = cls.ASSOCIATED_ARTIFACT_TYPES[0]
+            if not (
+                inspect.isclass(artifact_class)
+                and issubclass(artifact_class, BaseArtifact)
+            ):
                 raise MaterializerInterfaceError(
-                    f"Invalid materializer class '{name}'. When creating a "
-                    f"custom materializer, make sure to specify at least one "
-                    f"type in its ASSOCIATED_TYPES class variable.",
+                    f"ASSOCIATED_ARTIFACT_TYPES value '{artifact_class}' for "
+                    f"materializer '{name}' is not a `BaseArtifact` "
+                    f"subclass.",
                     url="https://docs.zenml.io/advanced-guide/pipelines/materializers",
                 )
+            artifact_type = artifact_class.TYPE_NAME
+
+        # Validate associated artifact type.
+        is_invalid_artifact_type = True
```

Comment:
I think this variable isn't needed, the `MaterializerInterfaceError` can simply be raised in the `except ValueError` clause right?